common/src/main/scala/org/apache/celeborn/common/rpc/RpcEndpoint.scala

```diff
@@ -134,6 +136,21 @@ trait RpcEndpoint {
       rpcEnv.stop(_self)
     }
   }
+
+  def checkAuth(context: RpcCallContext, appId: String): Unit = {
+    context match {
+      case remoteContext: RemoteNettyRpcCallContext =>
+        checkAuth(remoteContext.transportClient, appId)
+      case _ =>
+      // Do nothing if the context is not RemoteNettyRpcCallContext
+    }
+  }
+
+  private def checkAuth(client: TransportClient, appId: String): Unit = {
+    if (client.getClientId != null && !(client.getClientId == appId))
+      throw new SecurityException(
```

Comment:
`SecurityException` is typically for use by security manager in java (see `java.lang.SecurityManager`).
Might be better to create our own `Exception`, or perhaps use `IllegalStateException`

Reply:
Done.